package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +534,326 @@ def plot_qavg(self, filename=None, **kwargs):
+        """Calculate the persistence length for polymer chains
+
+        Parameters
+        ----------
+        u: Universe
+            trajectory
+        selection: tuple(string, string)
+            two contacting groups that change over time
+        refgroup: tuple(AtomGroup, AtomGroup)
+            two contacting groups in their reference conformation
+        radius: float, optional (4.5 Angstroms)
+            radius within which contacts exist
+        method: string
+            either 'cutoff' or 'best-hummer'
+
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+        Parameters for 'best-hummer' method
+        -----------------------------------
+        lambda_constant: float, optional (1.8 unitless)
+            contact is considered formed between (lambda*r0,r0)
+        beta: float, optional (5 Angstroms^-1)
+            softness of the switching function, the lower the softer
+
+        Attributes
+        ----------
+        results: list 
+            Fraction of native contacts for each frame
+        """
+
+        # check method
+        if not method in ("cutoff", "best-hummer"): 
+            raise ValueError("method has to be 'cutoff' or 'best-hummer'")
+        self.method = method
+
+        # method-specific parameters
+        if method == "best-hummer":
+            self.beta = kwargs.get('beta', 5.0)
+            self.lambda_constant  = kwargs.get('lambda_constant', 1.8)
+
+        # steup boilerplate
+        self.u = u
+        self._setup_frames(self.u.trajectory,
+                           start=start,
+                           stop=stop,
+                           step=step)
+
+        self.selection = selection
+        grA, grB = u.select_atoms(selection[0]), u.select_atoms(selection[1])
+        self.grA, self.grB = grA, grB        
+        refA, refB = refgroup
+
+        # contacts formed in reference
+        r0 = distance_array(refA.positions, refB.positions)
+        self.r0 = r0
+        self.mask = r0 < radius
+
+        self.contact_matrix = []
+        self.timeseries = []
+        self.outfile = outfile
+
+    def _single_frame(self):
+        grA, grB, r0, mask = self.grA, self.grB, self.r0, self.mask
+
+        # compute distance array for a frame
+        d = distance_array(grA.positions, grB.positions)
+
+        # select only the contacts that were formed in the reference state
+        # r, r0 are 1D array
+        r, r0 = d[mask], r0[mask]
+
+        if self.method == "cutoff":
+            y = r <= r0
+        elif self.method == "best-hummer":
+            y = 1/(1 + np.exp(self.beta*(r - self.lambda_constant * r0)))
+        else:
+            raise Exception("Unknown method type, has to be 'cutoff' or 'best-hummer'")
```

Comment:
Agree on the ValueError
Disagree on the `if else`: what if the user inadvertently changes the method attribute? 

I guess the way to do it is to make `method` into `_method` and then that implies 'don't-touch-thism' and we can do this check in __init__